src/routes/notifications/notifications.controller.spec.ts

```diff
@@ -0,0 +1,243 @@
+      const clientError = new NetworkResponseError(
+        faker.datatype.number({ min: 400, max: 499 }),
+      );
+      const serverError = new NetworkResponseError(
+        faker.datatype.number({ min: 500, max: 599 }),
+      );
+      mockNetworkService.post
+        .mockImplementationOnce(() => Promise.reject(clientError))
+        .mockImplementationOnce(() => Promise.reject(serverError))
+        .mockImplementation(() => Promise.resolve());
+
+      await request(app.getHttpServer())
+        .post('/register/notifications')
+        .send(registerDeviceDto)
+        .expect(500);
+    });
+
+    it('No status code errors returned from provider', async () => {
+      const chains: Chain[] = range(4).map((idx) =>
+        chainBuilder().with('chainId', idx.toString()).build(),
+      );
+      const safeRegistrations: SafeRegistration[] = chains.map((chain) =>
+        safeRegistrationBuilder().with('chain_id', chain.chainId).build(),
+      );
+      const registerDeviceDto = registerDeviceDtoBuilder()
+        .with('safeRegistration', safeRegistrations)
+        .build();
+      mockNetworkService.get.mockImplementation((url) => {
+        const getChainUrlPattern = `${safeConfigUrl}/api/v1/chains/`;
+        if (url.includes(getChainUrlPattern)) {
+          return Promise.resolve({ data: sample(chains) });
+        }
+        return Promise.reject(`No matching rule for url: ${url}`);
+      });
+      // return generic errors for two providers
+      mockNetworkService.post
+        .mockImplementationOnce(() => Promise.reject(new Error()))
+        .mockImplementationOnce(() => Promise.reject(new Error()))
+        .mockImplementation(() => Promise.resolve());
+
+      await request(app.getHttpServer())
+        .post('/register/notifications')
+        .send(registerDeviceDto)
+        .expect(500);
+    });
+
+    it('Other errors returned from provider', async () => {
+      const chains: Chain[] = range(4).map((idx) =>
+        chainBuilder().with('chainId', idx.toString()).build(),
+      );
+      const safeRegistrations: SafeRegistration[] = chains.map((chain) =>
+        safeRegistrationBuilder().with('chain_id', chain.chainId).build(),
+      );
+      const registerDeviceDto = registerDeviceDtoBuilder()
+        .with('safeRegistration', safeRegistrations)
+        .build();
+      mockNetworkService.get.mockImplementation((url) => {
+        const getChainUrlPattern = `${safeConfigUrl}/api/v1/chains/`;
+        if (url.includes(getChainUrlPattern)) {
+          return Promise.resolve({ data: sample(chains) });
+        }
+        return Promise.reject(`No matching rule for url: ${url}`);
+      });
+      // return redirection and generic errors for two providers
+      const redirection = new NetworkResponseError(
+        faker.datatype.number({ min: 300, max: 399 }),
+      );
+      mockNetworkService.post
+        .mockImplementationOnce(() => Promise.reject(redirection))
```

Comment:
Why is a redirection an error?

Reply:
True, in this case, it should be indicated in the client response, I will change this as well 👍

Reply:
I removed the redirection from the error scenarios in [0728521](https://github.com/5afe/safe-client-gateway-nest/pull/267/commits/0728521aa558fa73c90498d9c13798e313936c93)